code/game/objects/structures/trash_pile.dm

```diff
@@ -0,0 +1,102 @@
+/obj/structure/trash_pile
+	name = "trash pile"
+	desc = "A heap of garbage, but maybe there's something interesting inside?"
+	icon = 'icons/obj/trash_piles.dmi'
+	icon_state = "randompile"
+	density = TRUE
+	anchored = TRUE
+	layer = TABLE_LAYER
+	obj_flags = CAN_BE_HIT
+	pass_flags = LETPASSTHROW
+
+	max_integrity = 50
+
+	var/hide_item_time = 15
+
+	var/list/searchedby	= list()// Characters that have searched this trashpile, with values of searched time.
+
+/obj/structure/trash_pile/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/climbable)
+	icon_state = pick(
+		"pile1",
+		"pile2",
+		"pilechair",
+		"piletable",
+		"pilevending",
+		"brtrashpile",
+		"microwavepile",
+		"rackpile",
+		"boxfort",
+		"trashbag",
+		"brokecomp",
+	)
+
+/obj/structure/trash_pile/proc/do_search(mob/user)
+	if(contents.len) //There's something hidden
+		var/atom/A = contents[contents.len] //Get the most recent hidden thing
+		if (istype(A, /obj/item))
+			var/obj/item/I = A
+			to_chat(user,"<span class='notice'>You found something!</span>")
+			I.forceMove(src.loc)
+	else
+		//You already searched this one bruh
+		if(user.ckey in searchedby)
+			to_chat(user,"<span class='warning'>There's nothing else for you in \the [src]!</span>")
+		//You found an item!
+		else
```

Comment:
instead of this, return early after the warning to_chat() and unindent all of this